### DISCUSSION TITLE
Shutdown by remote Control

### DISCUSSION THREAD

**Discussion by iukbox** [hoylabs/OpenDTU-OnBattery#202] [Q&A] [answered]
Kommt dieser Eintrag im Ereignisprotokoll von der Basis DTU oder vom Power Limiter?
Ich versuche mich an einer Nulleinspeisung ohne Batterie. Funktioniert soweit alles gut ( empfohlene Einstellung vom anderen Thread übernommen ) bis auf die Meldung. Das untere Leistungslimit steht auf 50 Watt weil der WR anscheinend drunter abschaltet. Aber die Meldung gabs bisher noch nie ohne den Powerlimiter bzw. in der Basis DTU.
![image](https://user-images.githubusercontent.com/26631519/235189314-22a5927a-fd40-4104-b8d5-53bb747c1ab9.png)

**Comment by helgeerbe** [comment, marked as answer]
Das kommt darauf an. Jedenfalls hat die openDTU den Inverter ausgeschaltet. Entweder der PowerLimiter, per MQTT oder über die Web-Oberfläche im Live View.

**Comment by iukbox** [comment]
Dann wird's der Powerlimiter gewesen sein. MQTT ist noch nicht eingerichtet und per Hand von der Weboberfläche hab ich das nicht verursacht.
Dann hält sich er Powerlimiter nicht an seine Vorgaben. Min 50 Watt ist eingerichtet.

⁣BlueMail for Android herunterladen ​

**Comment by checkit99out** [comment]
Ich habe genau das gleiche Problem. Nutze ein Shelly 3EM. Die Leistungsdaten werden per MQTT über Homeassistent (Mosquitto-Broker) korrekt an die DTU gegeben.
Sobalod das "Powermeter" aktiviert ist, sehe ich auch in der DTU-Oberfläche die Summe der gesammten 3 Phasen.
Allerdings schaltet die DTU meinen Wechselrichter ab, selbst wenn ich  über 600 Watt an Verbraucher habe.
Der Homeassisten macht es nicht und manuell habe ich nichts eigegeben.
Möchte gerne eine "Nulleispeisung" (oder zumindest nicht mehhr als 150W über Eigenverbrauch), nutze aber keine Akkus oder Ladegeräte. Nur WR, DTU, HA.

KAnn mir jemand die notwendigen Einstellungen nennen?
(die Topics für die MQTT-VERbindung sind klar).
Danke

**Reply by helgeerbe** [reply]
Kurze Frage, habt ihr oberes Limit auf Max Leistung des Inverters eingestellt? Mir ist gestern etwas beim Testen aufgefallen. Wenn ich den Power Limiter auf 300 W als Limit setze (HM 300), dann schaltet er ab wenn 300W produziert werden sollen. Wenn ich 299 W als Limit eingebe, läuft alles,korrekt.

Muss das noch untersuchen.

**Comment by checkit99out** [comment]
@helgeerbe: kann ich wenig zu sagen, da das Peek meines WR bestimmt kaum erreicht wird, denn zu wenig Solarplatten (Wp) angeschlossen.
Aufgefallen ist mir allerdings:
Bei positiven Bezugswerten regelt die DTUon Battery. Sobald ein großer Verbraucher (z.B. Heizlüfter) abgeschaltet wird, wird der Bezugswert negativ - klar, Einspeisung in das Netz, denn ich habe ja keine Batterie am System.
Die Negativwerte werden jedoch nicht mehr geregelt. Es scheint, als wird nur der Wert - ohne Vorzeichen - beachtet.
Das führt natürlich dazu, dass statt 600 Watt für den Heizlüfter, nun 600W in das Netz gehen.
Ok, mit Batterie könnte man das verhindern, die lohnt sich für mich allerdings nicht.
Kann ich irgendwie einstellen, dass ein Negativwert (also Einspeisung in das Netz) weitgehend verhindert wird.

**Comment by helgeerbe** [comment]
Hm, ich weiss nicht ob ich dich richtig verstanden habe.

Eigentlich sollte der Power Limiter auf das Target regeln, wenn er es erreichen kann. Wenn target 0 W. Dann:

*  400 W am Panel. Oberes Limit 600 Watt, Verbrauch im Haus 1000W -> Inverter sollte auf 600 W stehen, es kommen nur 400W tatsächlich
* 400 W am Panel. Oberes Limit 600 Watt, Verbrauch im Haus 200 -> Inverter sollte auf 200 W stehen, normalerweise würde dann die Batterie mit 200 W geladen

Was hast du für einen PowerMeter? Werden die Daten richtig in der Konsole ausgegeben? Positiv bei Netzbezug, negativ wenn du ins Netzt einspeist?

**Reply by checkit99out** [reply]
Du hast richtig verstanden:
Powermeter ist ein Shelly 3EM, der die Daten mit MQTT an Mosquitto sendet. 
Die Werte in der Konsole der DTU passen. Positiv bei Netzbezug, Negativ bei Einspeisung.

"Target Power comsuption from grid" = 0
"Hysteriesis" = 20
"Lower Power Limit" =50
"upper power Limit" = 1499
"dynamic power limiter" = on
"enablle solar passthrough" = on.

Wie gesagt: Bei ausreichend Last regelt er. (Netztbezug vorhanden, größer oder kleiner, je nach Regelung)
Fällt die Last und erfolgt eine Einspiesung, regelt er nicht weiter. Eine positive/negative Null sollte ja eigentlich rauskommen.
Setup: WR direkt am Netz, DTU, keine Batterie, kein Laderegler für die Batterie.

**Reply by checkit99out** [reply]
habe das target auf NULL gestellt, hysteriesis auf 10, upper limit auf 1450, lower limit auf 10. Nun regelt er auch bei negativem Bezug (Einspeisung) auf 1% runter. 
Da mein Shelly 3EM in der Unterverteilung im Ruhebetrieb bei 9Watt misst, wäre das ok.
Teils schaltet die DTU leider den WR ab, evtl. weil der Verbrauch unter 10 W liegt (eben bei 9W) und damit unter lower limit?
Werde ich mal testen, wenn ein ńeuer Shelly im Zählerkasten direkt den Gesamtverbrauch des Hausen ink. der Unterverteilung misst. 

Kleinere, als 10W kann ich das lower limit nicht einstellen, dass lässt die DTU-GUI nicht zu.

**Comment by olomberg** [comment]
Moin Zusammen.
Bin aktuell auf	"d1885b6". Sehe, dass der Power Limiter immer noch sporadisch den Wechselrichter zwischendurch mal ausschaltet. Ist da noch ein Bug in der Berechnung? Danke.

**Reply by helgeerbe** [reply]
Bug kann immer sein.  Da müsste ich schon mehr wissen. 10W ist das niedrigste Limit, darunter bleibt nur abschalten. Kann also auch OK sein.

**Reply by olomberg** [reply]
HI.

Ich glaube ich habe die Ursache gefunden. Nutze powerfox Poweropti als Powermeter, welchen ich per HTTP + JSON abgreife. Leider ist die API von powerfox nicht so stabil. Habe gesehen, dass er 5-6 Mal auf einen TimeOut läuft und keine Werte bekommt. Damit schaltet dann der Dynamic Power Limiter der Wechselrichter aus. Kommt dann irgendwann mal wieder eine Wert wird er wieder angeschaltet. Das Ganze ist heute 15 - 20 Mal passiert. Plan B ist  jetzt die Daten per Telegraf abgreifen und in Mosquitto schreiben. Dann sollte sich das Thema erledigt haben. Nicht schön, aber ein Workaround. Leider kommt man nicht per LAN auf das Ding und muss die API nutzen.

**Comment by iukbox** [comment]
Heute mal abgewartet bis der Fall wieder eintritt. Einstellungen Power Limiter:
![image](https://user-images.githubusercontent.com/26631519/235342997-77f2ebe6-418d-4a5b-a8fd-96f3d5e7a02c.png)

![image](https://user-images.githubusercontent.com/26631519/235343612-75e68210-132c-4ecd-911b-8662249e5e16.png)
 Iich gehe nachher mal noch in die Influxdb um die Werte zu checken.

**Reply by iukbox** [reply]
Also an die Leistungsgrenze sind wir nicht gekommen. Diese Leistungszahl ist von 2 WR zusammen gerechnet. Einzelauswertung müßte ich einrichten wenn ihr es zum Debuggen braucht. Ist aber kein Thema. DB Zeit + 2 Stunden .....
![image](https://user-images.githubusercontent.com/26631519/235344006-3bce36fb-8752-4272-976c-52c0b0806083.png)

**Reply by helgeerbe** [reply]
Ich verstehe dein setup noch nicht. Wenn du mehrere Wechselrichter hast, und du so viel Strom erzeugst, dass openDtu weniger als 50W produzieren müsste, damit das Limit von 0Watt Import erreicht werden kann, dann wird abgeschaltet. 

Wichtig wäre zu wissen, wie viel Strom du Gerede aus dem Netz importierst, bzw. exportierst.

**Reply by iukbox** [reply]
Ich bin dann wahrscheinlich schon im Bereich der Einspeisung. Meine Annahme war das der WR dann auf das minimale Leistungslimit herunter geregelt wird, damit er nach Erhöhung der Abnahme aus dem Netz schneller wieder zur Verfügung steht.
Wenn das so gewollt ist das der unter 50 Watt abgeschaltet wird, dann ist das ja in Ordnung. Man muss es nur wissen :)

**Reply by helgeerbe** [reply]
@iukbox die minimale Leistung, die man am Inverter einstellen kann sind 10 W. Bei Batteriebetrieb ist es halt blöd, wenn dauerhaft 10W von der Batterie geholt werden, wenn dies eigentlich gar nicht nötig ist. Daher schalten wir den Inverter aus und erst wieder ein, wenn mehr als 10W erzeugt werden muss.

**Comment by spcqike** [comment]
ich vermute, ich hab das selbe Problem, die Diskussion hier aber gestern Abend nicht gefunden :)
#237

Nachdem ich bei der Batterievorgabe alle Spannungen auf 0 gesetzt habe, läuft der Inverter durch. (ich habe keine Batterie angeschlossen)

der Log sagt auch regelmäßig
12:29:00.435 > [PowerLimiterClass::loop] dcVoltage: 42.70 Voltage Start Threshold: 0.00 Voltage Stop Threshold: 0.00 inverter->isProducing(): 1

ich habs nicht getestet, aber vermutlich ist die DC Spannung irgendwo abweichend.
